app/core/utils.py

```diff
@@ -136,20 +147,75 @@ def _mode_to_result_and_type(
     return result_type, result
 
 
+connection_ttl_cache = TTLCache(maxsize=1, ttl=CONNECTION_STATUS_TTL_SECS)
+
+
+@cached(connection_ttl_cache)
+def is_connected() -> bool:
+    """Check if the system can establish a network connection to the test host. Result is cached for a short time."""
+    try:
+        socket.setdefaulttimeout(SOCKET_TIMEOUT)
```

Comment:
This will set the timeout globally for the process. Is that what you want? I think I'd just set a specific timeout value for the `socket.socket(...)` call.